### ISSUE TITLE
Tuple assignment inside loop breaks type inference

### ISSUE THREAD

**Issue by Molkree** [microsoft/pylance-release#2731]
## Environment data

-   Language Server version: v2022.4.2
-   OS and version: Win10
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.10

## Expected behavior

No errors.
## Actual behavior

> Type of "var" is partially unknown
  Type of "var" is "Unknown | int"

## Code Snippet / Additional information

```python
var = 0
while True:
    if var and True:  # no error if it's just `if var:`
        break
    else:
        var, _ = var + 1, 0
```
Regression in v2022.4.2.

I can reproduce it if the variable is changed in a tuple assignment and the variable is in RHS (doesn't matter if others are discards), if everything is inside the loop and if there's another `if` branch which can be executed.

**Comment by erictraut** [comment]
This is a core type checking issue, so I've created a tracking bug in the pyright repo.

As a simple workaround, you can annotate the type of `var` as `int`.

```python
var: int = 0
while True:
    if var and True:
        break
    else:
        var, _ = var + 1, 0
```

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by heejaechang** [comment]
This issue has been fixed in version 2022.4.3, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202243-27-April-2022)
